fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java

```diff
@@ -76,14 +75,6 @@ private static String analyzeEngineName(String engineName) {
             return EngineType.defaultEngine().name();
         }
 
-        if (engineName.equalsIgnoreCase(EngineType.STARROCKS.name()) && !Config.use_staros) {
-            throw new SemanticException("Engine %s needs 'use_staros = true' config in fe.conf", engineName);
-        }
-
-        if (engineName.equalsIgnoreCase(EngineType.OLAP.name()) && Config.use_staros) {
-            throw new SemanticException("Disallow create OLAP table in this cluster");
-        }
-
```

Comment:
whether need to keep this check?

reject create olap table in shared-data mode, reject create starrocks table in shared-nothing mode?

Reply:
not need, if user create olap table in shared-data mode, we will create a lake table for him.

Reply:
but the engine will still show as olap? should not do this until starrocks engine is completely removed.

Reply:
yes,  remove starrocks engine in this pr.

Reply:
Will remove starrrocks engine in this pr: https://github.com/StarRocks/starrocks/pull/18122/files